source/_integrations/switch.modbus.markdown

```diff
@@ -24,6 +24,7 @@ switch:
       slave: 1
       coil: 13
     - name: Switch2
+      hub: hub1
```

Comment:
Hub isn't a required option, and actually should be removed from all examples in the first example.

We try to keep all configuration samples minimal. Thus, no optional requirement in the default sample. This helps a user to get started quickly by copy-&-paste the sample without worrying about optional parameters which they most likely not need. If required, insert a full configuration sample later that covers special setups or alike.

Reply:
While hub might not be required in general, it is definitely required for `modbus`, I've had this issue - that's why I've created the pull request.
If you don't set hub name you will get [configuration error](https://community.home-assistant.io/t/modbus-component-cant-setup-platform/151156).

Reply:
According to the integration code, `hub` is optional and has a default value: <https://github.com/home-assistant/core/blob/dev/homeassistant/components/modbus/__init__.py#L83>

The documentation should reflect the integration code. If the integration handles it incorrect, that should be fixed. The documentation should not be used as a workaround.

Reply:
I'm not sure I'll be able to jump into the code and figure what's wrong, hopefully someone will jump in:)
But I totally agree with your points, let's discard this pull request than!